### ISSUE TITLE
Show age of pods in pod view

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#5587]
# Summary

Visual indicator of the age of each pod w/o having to click each pod and get details in pod view

# Motivation

We're watching nodes jump around. In the tree view, I can look at a deploy/sts/pod and see their age. In the new pod view, it's hard to know if a pod is new or old.

Specifically, we're performing a rollout of a stateful set, and some of the pods have been rolled out, and some haven't, and it'd be nice to be able to see at a glance which have/which haven't.

For `deployments` with `replicasets`, one can look at the revisions of the `replicasets` and actually watch as pods leave one and appear in the other. One can't do that in `statefulsets` -- there's only one set, and it just ages over time.

There actually is something called a `ControllerRevision` -- which if things were being nice would make this less painful, but they aren't part of the parentage model.

# Proposal

- Visit https://cd.apps.argoproj.io/applications/argo-events?view=pods

Note: while hovering is better than nothing, it'd be best if it were an indicator along the edge of the boxes.

Units [positions are selected for LTR orientations -- they (probably) could/should be mirrored for RTL under BiDi, but I don't think ArgoCD does any i18n, so that's a bridge to cross much later):
* minutes (`m`) (top left)
* hours (`h`) (top right)
* days (`D`) (bottom left)
* months (`M`)  (bottom right)
* years (`Y`) (bottom right)

## Details
* Lowercase is used for smaller things. Uppercase is used for bigger things.
* Younger (earlier) things are placed at the beginning of the imaginary box and progress right along the box, and then wrap to the beginning of the next line as they conceptually age.

Collectively these properties ensure that `minutes` and `Months` are not close to each other, giving them a much better chance of being distinguished.

- Yes, I know trying to use a box risks running into adjacent boxes, it's possible the boxes will need to grow a bit to accommodate. Note that even just using the position along the box will be enough most of the time to recognize older/younger.
- While it's possible to play with colors and shapes, I don't want to try playing with colors that until ArgoCD refreshes its color palette, and probably not even after, as I'm still waiting for the pods to gain a red for when they're being destroyed. 
- I don't think there's really enough space to place with shapes, although if someone wants to propose something that involves shapes, I'd be willing to give it a look.

I considered using the entire 360 degrees around the box and decided it's a bad choice -- degrees work when the 1 degree and 359 degree are meaningfully close together, `1m` is not close to `1M`.

**Screenshots**

Current:
![current](https://user-images.githubusercontent.com/2119212/108922732-86001b00-7605-11eb-8972-ce392713e12e.png)

Proposed
![with age indicators](https://user-images.githubusercontent.com/2119212/108924829-7387e080-7609-11eb-8396-326b8dd7cf4e.png)

**Comment by jessesuen** [comment]
Couple of us think it'll make the view too busy, but maybe in the tooltip?

**Comment by jsoref** [comment]
In the tooltip would be better than nothing (and could have a longer value), but it really wouldn't help because at a glance there are 5 or more dots and having to scan them (and remember the values of all the other ones) would be painful.

I'd almost have to ask for a shader application on the art if I couldn't get text.

Worst case is of course to add a shift-state to turn on this...

**Comment by glebk** [comment]
I think there is a consensus on adding the "created at" to the tooltip?

@jsoref am I correct in understanding that the main reason for having that information is for differentiation during a rollout? If so, can I suggest adding a simple visual indication, just for new pods, let's say younger than 30m, to help with this specifically? For example something like:
![image](https://user-images.githubusercontent.com/4512057/135766421-2ccacafe-e62e-477f-865d-654c867fcccf.png)

@alexmt if this one isn't worked on yet, can I take it?

**Comment by jsoref** [comment]
I'd go with a 🌟 as a visual indication if you're going that way. Yes, as long as we don't do try rolling restarts within your window, that'd address our use case.

**Comment by alexmt** [comment]
I agree that adding label with the age will make the view too busy. The visual indication sounds like a better option.

**Comment by reginapizza** [comment]
Is this still being worked on @glebk? If not, I'm interested and have some capacity this sprint to take it on. Please let me know!

**Comment by glebk** [comment]
@reginapizza please go ahead!
